franklin/github/views.py

```diff
@@ -49,5 +53,21 @@ def callback(request):
 
     return HttpResponse(status=200)
 
+@api_view(['POST'])
 def deploy_hook(request):
-    pass
+    if request.method == 'POST':
+        # Github webhooks contain special headers we can check for
+        event_type = request.META.get("HTTP_X_GITHUB_EVENT")
+        # Possibly useful for validation/security 
+        # request.META.get("HTTP_X_GITHUB_DELIVERY")
+
+        if event_type and event_type == "push":
+            # For now, we only support push to branch events
+            # TODO - can we check the branch name before doing work?
+            # master_branch = request.data.get("repository") #.master_branch
```

Comment:
can. will. One thing I was try to do was only accept changed for a specific branch (master). Was having trouble finding that in the json payload from github.